File: src/entities/GenreRating.ts
import { Column, Entity, JoinColumn, ManyToOne } from "typeorm";
import { Genre } from "./Genre";
import { Users } from "./Users";

@Entity("genre_rating", { schema: "dionysus" })
export class GenreRating {
  @Column("integer", { name: "rating", nullable: true })
  rating: number | null;

  @ManyToOne(() => Genre, (genre) => genre.genreRatings, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "genre_id", referencedColumnName: "genreId" }])
  genre: Genre;

  @ManyToOne(() => Users, (users) => users.genreRatings, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "user_id", referencedColumnName: "userId" }])
  user: Users;
}

File: engine/engine.api/Models/GenreRating.cs
using System;
using System.Collections.Generic;

namespace engine.api.Models
{
    public partial class GenreRating
    {
        public int? Rating { get; set; }
        public long? UserId { get; set; }
        public long? GenreId { get; set; }

        public virtual Genre? Genre { get; set; }
        public virtual User? User { get; set; }
    }
}
